src/documentation/pipelines/test-your-pipeline.md

```diff
@@ -5,4 +5,263 @@ permalink: /documentation/pipelines/test-your-pipeline/
+MyDoFn myDoFn = ...;
+DoFnTester<String, Integer> fnTester = DoFnTester.of(myDoFn);
+
+String testInput = "test1";
+List<Integer> testOutputs = fnTester.processBatch(testInput);
+```
+
+To check the results of `processBatch`, you use JUnit's `Assert.assertThat` method to test if the `List` of outputs contains the values you expect:
+
+```java  
+String testInput = "test1";
+List<Integer> testOutputs = fnTester.processBatch(testInput);
+
+Assert.assertThat(testOutputs, Matchers.hasItems(...));
+
+// Process a larger batch in a single step.
+Assert.assertThat(fnTester.processBatch("input1", "input2", "input3"), Matchers.hasItems(...));
+```
+
+## Testing Composite Transforms
+
+To test a composite transform you've created, you can use the following pattern:
+
+*   Create a `TestPipeline`.
+*   Create some static, known test input data.
+*   Use the `Create` transform to create a `PCollection` of your input data.
+*   `Apply` your composite transform to the input `PCollection` and save the resulting output `PCollection`.
+*   Use `PAssert` and its subclasses to verify that the output `PCollection` contains the elements that you expect.
+
+### Using the SDK Test Classes
+
+[TestPipeline](/https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/testing/TestPipeline.java) and [PAssert](http://beam.incubator.apache.org/documentation/sdks/javadoc/0.3.0-incubating/org/apache/beam/sdk/testing/PAssert.html) are classes included in the Beam Java SDK specifically for testing transforms.
+
+#### TestPipeline
+
+For tests, use `TestPipeline` in place of `Pipeline` when you create the pipeline object. Unlike `Pipeline.create`, `TestPipeline.create` handles setting `PipelineOptions` interally.
+
+You create a `TestPipeline` as follows:
+
+```java
+Pipeline p = TestPipeline.create();
+```
+
+#### PAssert
+
+`PAssert` is an assertion on the contents of a `PCollection`. You can use `PAssert` to verify that a `PCollection` contains a specific set of expected elements.
+
+For a given `PCollection`, you can use `PAssert` to verify the contents as follows:
+
+```java  
+PCollection<String> output = ...;
+
+// Check whether a PCollection contains some elements in any order.
+PAssert.that(output)
+.containsInAnyOrder(
+  "elem1",
+  "elem3",
+  "elem2");
+```
+
+Any code that uses `PAssert` must link in `JUnit` and `Hamcrest`. If you're using Maven, you can link in `Hamcrest` by adding the following dependency to your project's `pom.xml` file:
+
+```java 
+<dependency>
+    <groupId>org.hamcrest</groupId>
+    <artifactId>hamcrest-all</artifactId>
+    <version>1.3</version>
+    <scope>test</scope>
+</dependency>
+```
+
+For more information on how these classes work, see the [org.apache.beam.sdk.testing](http://beam.incubator.apache.org/documentation/sdks/javadoc/0.3.0-incubating/org/apache/beam/sdk/testing/package-summary.html) package documentation.
+
+### Using the Create Transform
```

Comment:
Feels like this should be before PAssert? So 
(1) create a test pipeline
(2) give it data
(3) hook up your composite
(4) passert the result